File: internal/router/option.go
package router

import (
	"uni-search-hub/internal/handler"
	"uni-search-hub/internal/middleware"

	"github.com/gin-gonic/gin"
)

func SetOptionRouter(router *gin.RouterGroup) {
	optionRouter := router.Group("/option")
	optionRouter.Use(middleware.RootAuth())
	{
		optionRouter.GET("/", handler.GetOptions)
		optionRouter.PUT("/", handler.UpdateOption)
	}
}

File: frontend/src/api/option.ts
import request from './request'
import type { ApiResponse, Option } from '../types'

// 获取系统配置
export function getOptions() {
  return request.get<ApiResponse<Option[]>>('/option/')
}

// 更新配置
export function updateOption(data: Option) {
  return request.put<ApiResponse>('/option/', data)
}
